lib/tasks/site_settings.rake

```diff
@@ -34,3 +34,53 @@ task "site_settings:import" => :environment do
 
   exit 1 if counts[:not_found] + counts[:errors] > 0
 end
+
+# Outputs a list of Site Settings that may no longer be in use
+desc "Find dead site settings"
+task "site_settings:find_dead" => :environment do
+  setting_names = SiteSettingsTask.names
+
+  setting_names.each do |n|
+    if !SiteSetting.respond_to?(n)
+      # Likely won't hit here, but just in case
+      puts "Setting #{n} does not exist."
+    end
+  end
+
+  directories = SiteSettingsTask.directories
+  dead_settings = []
+
+  if !SiteSettingsTask.rg_installed?
+    puts "Please install ripgrep to use this command"
+    exit 1
+  end
+
+  puts "Checking #{setting_names.count} settings in these directories:"
+  puts directories
+  puts
+
+  setting_names.each do |setting_name|
+    count = 0
+    count = SiteSettingsTask.rg_search_count("SiteSetting.#{setting_name}", directories[0])
```

Comment:
Total nitpick, but you could use rails magic `directories.first` :P

Reply:
lol, that it exactly what I was doing in the rails console playing around with this